sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -201,6 +202,7 @@ Connection getConnection() throws Exception {
   public abstract static class Read<T> extends PTransform<PBegin, PCollection<T>> {
     @Nullable abstract DataSourceConfiguration getDataSourceConfiguration();
     @Nullable abstract String getQuery();
+    @Nullable abstract List<Object> getParameters();
```

Comment:
I really like the idea of the `StatementPreparator` since it will allow us to provide a default one which just does what you suggest with using a list of java.sql.Type and if the user wants they can provide a different one if needed.